ballot/serializers/office_election.py

```diff
@@ -4,17 +4,36 @@
 from _django_utils.serializers import ExtendedModelSerializer
 
 
+class OfficeSerializer(serializers.ModelSerializer):
+    locality_id = serializers.PrimaryKeyRelatedField(source='locality', read_only=True)
+
+    class Meta:
+        model = models.Office
+        exclude = ('locality',)
+
+
 class CandidateSerializer(ExtendedModelSerializer):
     party = serializers.CharField(max_length=50, source='party.name')
+    ballot_item_id = serializers.PrimaryKeyRelatedField(source='ballot_item', read_only=True)
+    office_election_id = serializers.PrimaryKeyRelatedField(
+        source='office_election',
+        read_only=True)
 
     class Meta:
         model = models.Candidate
+        exclude = ('ballot_item', 'office_election')
 
 
-class OfficeElectionSerializer(ExtendedModelSerializer):
-    name = serializers.CharField(max_length=50, source='office.name')
+class OfficeElectionSerializer(serializers.ModelSerializer):
+    candidates = CandidateSerializer(many=True)
 
     class Meta:
         model = models.OfficeElection
-        exclude = ('contest_type',)
-        rename = dict(ballot='ballot_id')
+        exclude = ('ballot', 'contest_type', 'office')
+        rename = dict(locality='locality_id', ballot_item='ballot_item_id')
+
+    def to_representation(self, obj):
```

Comment:
Not a standard python thing, it's a [Django Rest Framework serializer](http://www.django-rest-framework.org/api-guide/serializers/#baseserializer) thing. You can customize the serializers with `to_representation` (serialize) and `to_internal_value` (deserialize).

From what I've seen with `__repr__`, it usually returns a string rather than a complex object like a `dict`.